handlers/client_test.go

```diff
@@ -0,0 +1,40 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestPayload(t *testing.T) {
+	require := require.New(t)
+
+	require.Equal(
+		"command=%7B%22addstream%22%3A%7B%22somestream%22%3A%7B%22source%22%3A%22http%3A%2F%2Fsome-storage-url.com%2Fvod.mp4%22%7D%7D%7D",
+		payloadFor(commandAddStream("somestream", "http://some-storage-url.com/vod.mp4")),
+	)
+	require.Equal(
+		"command=%7B%22push_start%22%3A%7B%22stream%22%3A%22somestream%22%2C%22target%22%3A%22http%3A%2F%2Fsome-target-url.com%2Ftarget.mp4%22%7D%7D",
+		payloadFor(commandPushStart("somestream", "http://some-target-url.com/target.mp4")),
+	)
+	require.Equal(
+		"command=%7B%22deletestream%22%3A%7B%22somestream%22%3A%7B%7D%7D%7D",
+		payloadFor(commandDeleteStream("somestream")),
+	)
+}
+
+// TODO: Remove after initial testing
+func TestWorkflow(t *testing.T) {
+	// first copy file into /home/Big_Buck_Bunny_1080_10s_1MB.mp4
+	mc := MistClient{apiUrl: "http://localhost:4242/api2"}
+	streamName, err := mc.AddStream("/home/Big_Buck_Bunny_1080_10s_1MB.mp4")
```

Comment:
This is actually a good point - we could potentially speed things up by doing the S3 download in Golang to local disk and handing over to Mist from there. 

We'd still want to eventually have Mist pull natively to avoid running out of disk space at scale, but this would get things working E2E with a bit less effort.

Reply:
yeah, I don't wee much difference if we do it from DMS API or Mist (as already implemented by @AlexKordic ). So, if Alex already implemented this part in Mist, then we can just pass the S3 URL to Mist.

Reply:
Isn't the upload part covered by https://github.com/DDVTECH/mistserver/pull/66 ? Afaik with that merged or cherry picked on your branch you can use S3 as a stream source directly

Reply:
Ok, sounds good!

Reply:
LMK if you run into any issues, it might require some config on the input side to recognize that s3:// URL's are supported